### PR TITLE
SPESS PARALLAX

### PR DESCRIPTION
**Pull request by monster860**
:cl: monster860
rscadd: Adds space parallax.
rscadd: Adds launch animations to shuttles using parallax.
/:cl:

Ported from https://github.com/FTL13/FTL13

(Click on the gifs to view full size)

[Parallax](https://gfycat.com/SatisfiedWaterloggedEchidna)
[Launch](https://gfycat.com/FreshDrearyKoodoo)

### CODE REVIEW

File: code/controllers/subsystem/mobs.dm [hunk 1]
```diff
@@ -30,6 +30,12 @@ var/datum/subsystem/mobs/SSmob
 		currentrun.len--
 		if(M)
 			M.Life(seconds)
+			// Hijacking the mobs subsystem for parallax. It's less laggy this way.
+			// I know this is a bit hacky, but all other options either break when
+			// you have a situation like someone in a locker, or a drone in someone's
+			// hand in a locker, or both.
```
Comment (monster860) [inline code/controllers/subsystem/mobs.dm:36]: Damn, I didn't finish my statement in the comment. Other methods are either that, or laggy, or both.

**Comment by oisin100** [comment]
YES! Finally.

**Comment by matskuman5** [comment]
It looks like all those stars are just a few hundred meters away seeing how strong the parallax is.

**Comment by AndySok** [comment]
it looks so.

Beautiful

File: code/game/atoms_movable.dm [hunk 2]
```diff
@@ -16,6 +16,7 @@
 	var/inertia_dir = 0
 	var/pass_flags = 0
 	var/moving_diagonally = 0 //0: not doing a diagonal move. 1 and 2: doing the first/second step of the diagonal move
+	var/list/mobs_in_contents = list() // This contains all the client mobs within this container
```
Comment (Cruix) [inline code/game/atoms_movable.dm]: Get rid of this and just do a for(var/mob/M in contents) whenever you want to loop through them. having a var initialized to a list in /atom/movable will noticeably slow down game initialization.

File: code/game/atoms_movable.dm [hunk 3]
```diff
@@ -132,6 +135,7 @@
 				continue
 			AM.Crossed(src)
 		Moved(oldloc, 0)
+		update_parallax_contents()
```
Comment (Cruix) [inline code/game/atoms_movable.dm]: Move this to Moved() as well

File: code/game/atoms_movable.dm [hunk 4]
```diff
@@ -78,6 +79,8 @@
 
 	if(. && has_buckled_mobs() && !handle_buckled_mob_movement(loc,direct)) //movement failed due to buckled mob(s)
 		. = 0
+	
+	update_parallax_contents()
```
Comment (Cruix) [inline code/game/atoms_movable.dm]: Move this to Moved() since both Move() and forceMove() call it, since they both also call Moved()

**Comment by monster860** [comment]
Alright, I've made it so there isn't the overhead of initializing movable atoms with a list.

**Comment by Time-Green** [comment]
MERGE IT ALREADY!!

**Comment by Super3222** [comment]
My only criticism is that the shuttle looks like it's moving a bit slow compared to hyperspeed fast.

**Comment by Cruix** [comment]
Looks good now, well done.

**Pull request merged**